### ISSUE TITLE
Multi-Ldap Authentication for Agents Working, but not for Clients

### ISSUE THREAD

**Issue by BigMattR** [philbertphotos/osticket-multildap-auth#34]
Attempting to change over from the original Ldap / Active Directory Plugin you can get for OSTicket to @jphilbert 's 1,2,2 Multi-Ldap as we need to be able to authenticate multiple other child domains beyond the one the server lives in.

We're running 1.12.3 on a windows 2016 Server, the regular auth-ldap.phar as stated was what was originally in use, and we're trying to change over.

I've gotten @jphilbert 's 1.2.2 to work with Agents with the single first domain (not adding additional until I can get it fully functioning with one), but the client login portal fails to work.

The log entries in the ost_logs table only show a search and info poll being done, even with debug and verbose enabled, but nothing to indicate why the client side fails, but the agent side works.

| 603 | Debug | ldap info (sys_eolsgenfsro) | [] | | 10.166.35.117 | 2020-02-26 20:55:57 | 2020-02-26 20:55:57 |
| 602 | Debug | ldap search (sys_eolsgenfsro) | [] | | 10.166.35.117 | 2020-02-26 20:55:57 | 2020-02-26 20:55:57 |

**Comment by philbertphotos** [comment]
I have been away ... can you tell me if this is still happening.?

**Comment by ScottishSamurai** [comment]
Not sure if this is related but the sync doesnt seem to be working for me. When I got to create a ticket it will pull the person from the ldap but not fill in the information. It also does not force registration. I believe that the system may be seeing something because when I try to log in with the users information I get the message "Access Denied. Contact your help desk administrator to have an account registered for you" but if I try to put in a random password for the user it just says "Access Denied"

**Comment by philbertphotos** [comment]
@ScottishSamurai do you get any errors in email? Database errors? What bout the logs when it syncs.
I have my system set to "public Anyone can register" I have to see if it works in private or disable mode.

Send me any errors or logs you may receive.

**Comment by ScottishSamurai** [comment]
I do have mine on as private. Originally when I sent you the errors i was getting in the other tickets I was getting emails. After the fix I have not received any even though I didn't change any of the settings. 

Logs are showing authenticating however if you look at the users it doesn't show them as registered. and still gives that error.
ldap login (username[client])
User authenticated on (domainname)

 ldap session (Username)
{"session":{"session_id":"80a4b4c5ce75b59f9a0d59627d5f2e84", "userID":37, "browser":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/86.0.4240.75 Safari\/537.36", "ip":"---------", "validated":false}, "token":null}

Also after the fix for adding the table ldap_sync I dont see any data getting added to it. Wasnt sure if that is suppose to be that way or not.

**Comment by philbertphotos** [comment]
Let me do a few test and get back to you in a few hours/days. My issue is that the plugin works for me for the most part for years but I really want to polish it up so it can works flawlessly for others with the added features I have in mind. 
Data will be added to in in a next update it was in beta and never fully fleshed out... it updates users based solely on the objectID this will fix problems when a user changes their name. If not I would have to manual change they username in Osticket to avoid SQL errors this will automate the process.

**Comment by philbertphotos** [comment]
Actually looking at mines you should see data in it if the syncs are working properly or at least running. Do you see "LDAP Sync" in the logs?

**Comment by ScottishSamurai** [comment]
I dont show any logs for sync, I just see logs for when it does Ldap look up when I search for a name when opening a ticket. 

I do see this for the Sync.
Next schedule: {"id":172,"namespace":"plugin.3","key":"sync_mailfrom","value":2,"updated":"2020-10-16 17:46:28"}

Are the Reset and Sync Users buttons suppose to do anything?

**Comment by philbertphotos** [comment]
Ok making some major changes in the config.php and minor ones in the sync_ldap and adding some debug stuff so by tomorrow it will be release currently testing stuff.

**Comment by ScottishSamurai** [comment]
Not sure how I did anything different but I did uncheck, Enable agent synchronization and I started getting the emails again about errors. I saw that one of the error showed that line 263 had the table "ost_user" hard coded instead of using ". TABLE_PREFIX ."
 Most of my users have now be automatically moved to Registered but not all. I am reviewing to see if I can find the users differences. I was able to log in as one of the users after they got registered. So I looks like once users are auto registered they can auth against the LDAP.

**Comment by philbertphotos** [comment]
Let me look at that ... I thought I removed all that crap

**Comment by philbertphotos** [comment]
@ScottishSamurai the system will not create accounts if client registration is set to "Private or Disabled"

**Comment by ScottishSamurai** [comment]
@philbertphotos Thank you for your help with all this! I am not creating new accounts at the moment. The ones in system that already are created I was needed to have them registered so users can login. Your plugin registered most of them so that has been a great help. I am reviewing the ones that didn't get registered to see what is different about them. 

I was able to figure out where part of my problems have been. I didn't know that the sync_mldap.php from the multi-ldap folder was also in the scp folder. Not sure if it is suppose to be this way or was something I did wrong. When I updated the file in the multi-ldap folder I was still getting the same errors. After I updated the other file the errors stopped. I noticed that the update in your code still had an issue on line 263.

Current:
$sql_value = db_result(db_query("SELECT name FROM " . TABLE_PREFIX . "ost_user	

Fix:
$sql_value = db_result(db_query("SELECT name FROM " . TABLE_PREFIX . "user
